FLChess/src/GameFrame.java

```diff
@@ -313,6 +319,14 @@ public void handleSelection(TilePanel newTile)
 
             // save selected tile
             selectedTile = newTile;
+            
+          /*if the selected tile color does not match the color of the player's turn,
+            return an error message */
+            if(selectedTile.pieceAt.white != GameFrame.whiteTurn) { 
+                System.out.println("Not your turn");

```

Comment:
I think if you add 'selectedTile = NULL_TILE;' immediately after the 'Not your turn' print statement, the weird selectedTile bug that happens after you click a piece its not the turn of would be fixed.